bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageItemImpl.java

```diff
@@ -0,0 +1,73 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2020 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v2;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.jetbrains.annotations.NotNull;
+
+import com.adobe.cq.wcm.core.components.models.PageItem;
+
+public class PageItemImpl implements PageItem {
+
+    String prefixPath;
+    ValueMap properties;
+    Element element;
+    Location location;
+    Map<String, String> attributes;
+
+    public PageItemImpl(@NotNull String prefixPath, @NotNull Resource resource) {
+        this.prefixPath = prefixPath;
+        this.properties = resource.getValueMap();
+    }
+
+    @Override
+    public Element getElement() {
+        if (element == null) {
+            element = Element.fromString(properties.get(PageItem.PROP_ELEMENT, String.class));
+        }
+        return element;
+    }
+
+    @Override
+    public Location getLocation() {
+        if (location == null) {
+            location = Location.fromString(properties.get(PageItem.PROP_LOCATION, String.class));
+        }
```

Comment:
I was able to place a `META` tag outside of `HEAD` section.  I guess `location` property should be ignored for `META` and `LINK` elements as they are only allowed/make sense in `HEAD` section.

Reply:
You're right about `meta`, it's a `head`-only element. `link` can occur in `body` too:

> A `<link>` element can occur either in the `<head>` or `<body>` element, depending on whether it has a link type that is body-ok. For example, the stylesheet link type is body-ok, and therefore `<link rel="stylesheet">` is permitted in the body. However, this isn't a good practice to follow; it makes more sense to separate your `<link>` elements from your body content, putting them in the `<head>`. (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link)

I would allow this to be controlled by developers though. There are some creative CSS hacks with `meta` tags out there ;)

Reply:
I stand corrected :) , its not recommended but allowed your right. As for meta, if thats the case with CSS hacks , then yes lets keep it in control of developers